### ISSUE TITLE
CDI 2.0: An injected security Principle does not work in a multi-threaded application

### ISSUE THREAD

**Issue by tevans78** [OpenLiberty/open-liberty#1436]
Given this example class...
```
@RequestScoped
public class MultiThreadCDIBean {
    @Inject
    private Principal user;
    @Resource
    private ManagedScheduledExecutorService executor;

    public String getName() {
        return getName(false);
    }

    public String getName(boolean supressException) {
        try {
            return executor.submit(new Callable<String>() {

                @Override
                public String call() throws Exception {
                    return user.getName();
                }
            }).get();
        } catch (InterruptedException e) {
            if (!supressException) {
                e.printStackTrace();
            }
        } catch (ExecutionException e) {
            if (!supressException) {
                e.printStackTrace();
            }
        }
        return "World!";
    }
}
```

When user.getName() is called on the managed executor thread, an exception occurs...
```
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R Caused by: 
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R org.jboss.weld.exceptions.NullInstanceException: WELD-000044: Unable to obtain instance from org.jboss.weld.bean.builtin.ee.PrincipalBean$PrincipalCallable@87f7658e
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at org.jboss.weld.bean.builtin.CallableMethodHandler.invoke(CallableMethodHandler.java:40)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at org.jboss.weld.bean.proxy.EnterpriseTargetBeanInstance.invoke(EnterpriseTargetBeanInstance.java:56)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:106)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at org.jboss.weld.security.Principal$$Proxy$_$$_Weld$Proxy$.getName(Unknown Source)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at com.ibm.cdi.test.cdisecurity.MultiThreadCDIBean$1.call(MultiThreadCDIBean.java:40)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at com.ibm.cdi.test.cdisecurity.MultiThreadCDIBean$1.call(MultiThreadCDIBean.java:36)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at com.ibm.ws.threading.internal.PolicyTaskFutureImpl.run(PolicyTaskFutureImpl.java:685)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at com.ibm.ws.threading.internal.PolicyExecutorImpl.runTask(PolicyExecutorImpl.java:1128)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     at com.ibm.ws.threading.internal.PolicyExecutorImpl$GlobalPoolTask.run(PolicyExecutorImpl.java:186)
[04/01/18 14:53:33:695 GMT] 0000006a SystemErr                                                    R     ... 3 more
```

**Comment by neilgsyoung** [comment]
I was able to recreate the issue pretty immediately with CDI2.0 and the test runs as it should when I switch to CDI 1.2.

What I found, after delving into Declarative Services, that in my CDI2.0 environment I was missing a key bundle - com.ibm.ws.cdi.security.

This issue was due to a feature dependency. My server.xml specified cdi-2.0 and appSecurity-2.0. If I switched to appSecurity-3.0 then my "missing" bundle was installed and the test code worked. The dependency on appSecurity-3.0 is described in com.ibm.websphere.appserver.cdi2.0-appSecurity1.0.feature. This internal feature pulls in the missing bundle.
